massa-protocol-worker/src/connectivity.rs

```diff
@@ -257,87 +257,103 @@ pub(crate) fn start_connectivity_thread(
                         let mut slot_default_category = config.default_category_info.target_out_connections.saturating_sub(peers_connected.iter().filter(|(_, peer)| {
                             peer.1 == PeerConnectionType::OUT && peer.2.is_none()
                         }).count());
-                        let mut addresses_to_connect: Vec<SocketAddr> = Vec::new();
+
+                        // Get all the addresses we can cannot to, without any filter or priorisation done yet
+                        let mut addresses_can_connect  = Vec::new();
                         {
                             let peer_db_read = peer_db.read();
-                            for (_, peer_id) in &peer_db_read.index_by_newest {
+                            for (peer_id, peer_info) in &peer_db_read.peers {
```

Comment:
to improve this part, maybe it's possible to calculate the number of connections needed before this loop and avoid to iter over all peer_db_read.peers. 
No need to iter 1000 times if we want only one connection

Reply:
Yes but as we need to prioritize peers, in the case we need only `n` connections we have to consider all the peers and take the `n` best peers.
If you break the loop earlier, you won't see all the peers, and may miss some with higher priority

Reply:
Yes, you are right